easybuild/easyconfigs/p/PETSc/PETSc-3.17.0-foss-2021a.eb

```
@@ -0,0 +1,84 @@
+# Updated to version 3.17.0 with significantly more modules added
+# J. Sassmannshausen ICL/UK 
+
+easyblock = 'ConfigureMake'
+
+name = 'PETSc'
+version = '3.17.0'
+
+homepage = 'https://www.mcs.anl.gov/petsc'
+description = """PETSc, pronounced PET-see (the S is silent), is a suite of data structures and routines for the
+ scalable (parallel) solution of scientific applications modeled by partial differential equations.
+
+This installation is build with significant amounts of modules in it compared to previous versions."""
+
+toolchain = {'name': 'foss', 'version': '2021a'}
+toolchainopts = {'openmp': True, 'usempi': True, 'pic': True}
+
+source_urls = [
+    'https://ftp.mcs.anl.gov/pub/petsc/release-snapshots/',
+    'ftp://ftp.mcs.anl.gov/pub/petsc/release-snapshots/',
+]
+sources = [SOURCELOWER_TAR_GZ]
+patches = [
+    'webclient-3.17.0.patch',
+]
+checksums = [
+    '96d5aca684e1ce1425891a620d278773c25611cb144165a93b17531238eaaf8a',  # petsc-3.17.0.tar.gz
+    '2ba253df42853385e2683dbc87dfcb66d4ce4bcd84e054efa8f466274f49bc81',  # webclient-3.17.0.patch
+]
+
+builddependencies = [('CMake', '3.20.1')]
+
+dependencies = [
+    ('Python', '3.9.5'),
+    ('SciPy-bundle', '2021.05'),
+    ('Boost', '1.76.0'),
+    ('METIS', '5.1.0'),
+    ('SCOTCH', '6.1.0'),
+    ('MUMPS', '5.4.0', '-metis'),
+    ('SuiteSparse', '5.10.1', '-METIS-5.1.0'),
+    ('Hypre', '2.21.0'),
+    ('zlib', '1.2.11'),
+    ('BLIS', '0.8.1'),
+    ('Szip', '2.1.1'),
+    ('HDF5', '1.12.1'),
+    ('Eigen', '3.3.9'),
+    ('GMP', '6.2.1'),
+    ('GSL', '2.7'),
+    ('libjpeg-turbo', '2.0.6'),
+    ('libpng', '1.6.37'),
+    ('giflib', '5.2.1'),
+    ('OpenSSL', '1.1', '', True),
+    ('libyaml', '0.2.5'),
+    ('zstd', '1.4.9'),
+    ('Mmg', '5.6.0'),
+    ('MPFR', '4.1.0'),
+    ('googletest', '1.11.0'),
+    ('muParser', '2.3.3'),
+    ('ParMETIS', '4.0.3'),
+]
+
+# enabling --with-mpi4py seems to be totally broken, leads to make errors like:
+# No rule to make target 'mpi4py-build'
+configopts = '--configModules=PETSc.Configure --optionsModule=config.compilerOptions '
+configopts += '--LIBS="-lm -lpthread -lrt" --with-mpi4py=0 '
+configopts += '--CFLAGS="-O2 -ftree-vectorize -march=native -fno-math-errno -fopenmp -fPIC" '
+configopts += '--CXXFLAGS="-O2 -ftree-vectorize -march=native -fno-math-errno -fopenmp -fPIC '
+configopts += '-DOMPI_SKIP_MPICXX -DMPICH_SKIP_MPICXX" --with-mpi=1 --with-build-step-np=4 '
+configopts += '--FFLAGS="-O2 -ftree-vectorize -march=native -fno-math-errno -fopenmp -fPIC" '
+configopts += '--with-shared-libraries=1 --with-debugging=0 --with-pic=1 --with-x=0 '
+configopts += '--with-windows-graphics=0 --with-numpy=1 --with-fftw=1 --with-scalapack=1 '
+configopts += '--with-blaslapack-lib="[/home/sassy/apps/software/FlexiBLAS/3.0.4-GCC-10.3.0/lib/'
+configopts += 'libflexiblas.a,libgfortran.a]" '
```

this could do with cleaning up, probably something along the lines of:
```suggestion
configopts += '--LIBS="$LIBS -lrt" --with-mpi4py=0 '
configopts += '--CFLAGS="$CFLAGS" '
configopts += '--CXXFLAGS="$CXXFLAGS '
configopts += '-DOMPI_SKIP_MPICXX -DMPICH_SKIP_MPICXX" --with-mpi=1 --with-build-step-np=4 '
configopts += '--FFLAGS="$FFLAGS" '
configopts += '--with-shared-libraries=1 --with-debugging=0 --with-pic=1 --with-x=0 '
configopts += '--with-windows-graphics=0 --with-numpy=1 --with-fftw=1 --with-scalapack=1 '
configopts += '--with-blaslapack-lib="$BLAS_LAPACK_LIB_DIR"
'
```

---

Thanks for the suggestion. much appreciated. The "$BLAS_LAPACK_LIB_DIR" is actually not needed, what is needed are the two libs `libflexiblas.a,libgfortran.a` which I replaced with `-lflexiblas -lgfortran` to stick to dynamic libs. 
Please have a look at the log file to make sure I am not accidentally build it with the wrong flags. From what I can see, it is all working as intended but that does not mean it is correct.